File: 20/qualification/A_Vestigium/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	input := bufio.NewScanner(os.Stdin)

	t := readInt(input)
	for tCase := 0; tCase < t; tCase++ {
		n := readInt(input)

		matrix := make([][]int, n)
		for r := 0; r < n; r++ {
			matrix[r] = readRow(input)
		}

		k, r, c := compute(matrix, n)

		fmt.Printf("Case #%d: %d %d %d\n", tCase+1, k, r, c)
	}
}

func compute(matrix [][]int, n int) (int, int, int) {

	// trace
	var k int
	for i := 0; i < n; i++ {
		k += int(matrix[i][i])
	}

	// count dup rows
	var r int
	for i := 0; i < n; i++ {
		m := make(map[int]struct{})
		for j := 0; j < n; j++ {
			m[matrix[i][j]] = struct{}{}
		}

		if len(m) != n {
			r++
		}
	}

	// count dup cols
	var c int
	for j := 0; j < n; j++ {
		m := make(map[int]struct{})
		for i := 0; i < n; i++ {
			m[matrix[i][j]] = struct{}{}
		}

		if len(m) != n {
			c++
		}
	}

	return k, r, c
}

func readInt(input *bufio.Scanner) int {
	input.Scan()
	n, err := strconv.Atoi(strings.TrimSpace(input.Text()))
	if err != nil {
		log.Fatal(err)
	}
	return n
}

func readRow(input *bufio.Scanner) []int {
	input.Scan()
	line := strings.TrimSpace(input.Text())
	sList := strings.Split(line, " ")

	intList := make([]int, len(sList))
	for i := range sList {
		n, err := strconv.Atoi(sList[i])
		if err != nil {
			log.Fatal(err)
		}
		intList[i] = n
	}
	return intList
}

File: 20/qualification/A_Vestigium/main.py
from sys import stdin


def trace(matrix):
    n = len(matrix)

    s = 0
    for i in range(n):
        s += matrix[i][i]
    return s


def duplicated_rows(matrix):
    dups = 0
    for row in matrix:
        if len(row) != len(set(row)):
            dups += 1
    return dups


def duplicated_columns(matrix):
    dups = 0
    for j in range(len(matrix)):
        column = [row[j] for row in matrix]
        if len(column) != len(set(column)):
            dups += 1
    return dups


def compute(matrix):
    k = trace(matrix)
    r = duplicated_rows(matrix)
    c = duplicated_columns(matrix)
    return k, r, c


def main():
    t = int(stdin.readline().strip())

    for t_i in range(t):

        N = int(stdin.readline().strip())

        matrix = []
        for _ in range(N):
            row = [int(x) for x in stdin.readline().strip().split()]
            matrix.append(row)

        k, r, c = compute(matrix)

        print(f"Case #{t_i + 1}: {k} {r} {c}")


main()
